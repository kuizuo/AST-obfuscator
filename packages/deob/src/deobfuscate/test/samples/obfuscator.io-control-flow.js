function a() {
  const h = [
    '4496112hRzUwi',
    'convertESM',
    'replaceRequireCalls',
    'inlineVarInjections',
    '10034958BKGbbp',
    'modules',
    'capture',
    '4|3|0|5|2|6|1',
    'aokfw',
    'convertDefaultRequire',
    'numericLiteral',
    '596381RBbCwG',
    'identifier',
    '6506843sEFStO',
    'callExpression',
    '220wEDVBy',
    'forEach',
    '270594BMxBNt',
    '1730kqELtV',
    'split',
    '6KqUKoX',
    'HLAuI',
    'require',
    '135748VvmqOB',
    '13597944MAVUeU',
  ]
  a = function () {
    return h
  }
  return a()
}
(function (c, d) {
  const e = c()
  while ([]) {
    try {
      const f = -Number.parseInt(b(0x15)) / 0x1 * (Number.parseInt(b(0x5)) / 0x2) + Number.parseInt(b(0xA)) / 0x3 + Number.parseInt(b(0x8)) / 0x4 * (-Number.parseInt(b(0x0)) / 0x5) + -Number.parseInt(b(0xE)) / 0x6 + Number.parseInt(b(0x17)) / 0x7 + -Number.parseInt(b(0x9)) / 0x8 + Number.parseInt(b(0x2)) / 0x9 * (Number.parseInt(b(0x3)) / 0xA)
      if (f === d) {
        break
      }
      else {
        e.push(e.shift())
      }
    }
    catch (g) {
      e.push(e.shift())
    }
  }
}(a, 0xEE0D8))
function b(c, d) {
  const e = a()
  b = function (f, g) {
    f = f - 0x0
    const h = e[f]
    return h
  }
  return b(c, d)
}
function applyTransforms() {
  const c = {
    HLAuI: b(0x11),
    aokfw: b(0x7),
  }
  const d = c[b(0x6)][b(0x4)]('|')
  let e = -0x1A70 + 0x93D + 0x275 * 0x7
  while ([]) {
    switch (d[e++]) {
      case '0':
        (0x0, getDefaultExport_1[b(0x13)])(this)
        continue
      case '1':
        return g
      case '2':
        var f = m[b(0x10)](m[b(0x14)]())
        continue
      case '3':
        this[b(0xF)][b(0x1)](esm_1[b(0xB)])
        continue
      case '4':
        this[b(0xF)][b(0x1)](varInjection_1[b(0xD)])
        continue
      case '5':
        this[b(0xC)]()
        continue
      case '6':
        var g = m[b(0x18)](m[b(0x16)](c[b(0x12)]), [f])
        continue
    }
    break
  }
}
