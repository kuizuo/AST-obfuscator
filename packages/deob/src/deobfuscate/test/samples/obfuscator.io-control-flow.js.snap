function applyTransforms() {
  this.modules.forEach(varInjection_1.inlineVarInjections);
  this.modules.forEach(esm_1.convertESM);
  (0, getDefaultExport_1.convertDefaultRequire)(this);
  this.replaceRequireCalls();
  var f = m.capture(m.numericLiteral());
  var g = m.callExpression(m.identifier("require"), [f]);
  return g;
}